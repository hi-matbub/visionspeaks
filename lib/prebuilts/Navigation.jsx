import React from 'react';
import PropTypes from 'prop-types';
import Navbar from 'react-bootstrap/Navbar';
import Nav from 'react-bootstrap/Nav';
import Cart from './Cart';

const Navigation = ({ cart, data, handleUpdatedCartInState, currentBill }) => (
  <div
    className="d-flex justify-content-between w-100 fixed-top"
    style={{ background: '#FFFFFF !important' }}
  >
    <Navbar
      expand="lg"
      style={{ backgroundColor: '#FFFFFF !important' }}
      fixed="top"
      className="justify-content-between fixed-top"
    >
      <img
        src={data.store.logo}
        height="75"
        className="d-inline-block align-top"
        alt="React Bootstrap logo"
      />
      <Navbar.Brand href="/">{data.store.name}</Navbar.Brand>
      <Nav>
        <Cart
          cart={cart}
          data={data}
          currentBill={currentBill}
          handleUpdatedCartInState={(updatedCart) =>
            handleUpdatedCartInState(updatedCart)
          }
        />
      </Nav>
    </Navbar>
  </div>
);

Navigation.propTypes = {
  data: PropTypes.shape({
    store: PropTypes.shape({
      name: PropTypes.string.isRequired,
      logo: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
  cart: PropTypes.shape({}),
  handleUpdatedCartInState: PropTypes.func.isRequired,
  currentBill: PropTypes.number,
};

Navigation.defaultProps = {
  cart: {},
  currentBill: 0,
};

export default Navigation;
