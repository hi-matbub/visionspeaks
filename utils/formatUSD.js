/* eslint-disable import/prefer-default-export */
const formatUSD = (number) => {
  // for inconsistent types
  const string = JSON.stringify(parseFloat(number));
  const { length } = string || number;
  const str = `${string.substring(0, length - 2)}.${string.substring(
    length - 2
  )}`;
  return parseFloat(str).toFixed(2);
};

export { formatUSD };
