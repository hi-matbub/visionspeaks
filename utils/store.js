exports.store = {
  name: 'Dog-Gone Goods',
  logo: '/logo.svg',
  hero: '/hero.svg',
  url: '',
  description: 'A dog treat shop',
  about: 'Everyone loves a happy dog!',
  contact: 'youremail@here.com',
  social: {
    facebook: '',
    twitter: '',
    instagram: '',
  },
};
