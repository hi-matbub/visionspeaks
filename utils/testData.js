// used for testing purposes

exports.testData = {
  store: {
    name: 'Dog-Gone Goods',
    logo: '/logo.svg',
    hero: '/hero.svg',
    url: '',
    description: 'A dog treat shop',
    about: 'Everyone loves a happy dog!',
    contact: 'youremail@here.com',
    social: { facebook: '', twitter: '', instagram: '' },
  },
  products: [
    {
      id: 'prod_IRbpyL6fgse3n',
      object: 'product',
      active: true,
      attributes: [],
      created: 1606151601,
      description: 'demo product description',
      images: ['img.jpg'],
      livemode: false,
      metadata: {
        price: 287,
      },
      name: 'Large Demo Product',
      statement_descriptor: null,
      type: 'service',
      unit_label: null,
      updated: 1606153202,
    },
    {
      id: 'prod_IRbouvicOUom16',
      object: 'product',
      active: true,
      attributes: [],
      created: 1606151551,
      description: 'demo product description',
      images: ['img.jpg'],
      livemode: false,
      metadata: {
        price: 287,
      },
      name: 'Demo Product',
      statement_descriptor: null,
      type: 'service',
      unit_label: null,
      updated: 1606153221,
    },
  ],
};
