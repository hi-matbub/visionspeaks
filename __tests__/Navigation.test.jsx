import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Navigation from '../lib/prebuilts/Navigation';
import { testData } from '../utils/testData';

const handleUpdatedCartInState = () => {}

it('Navigation component renders smoke test without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Navigation 
      data={testData} 
      handleUpdatedCartInState={handleUpdatedCartInState} 
    />
  , div);
  ReactDOM.unmountComponentAtNode(div);
});

it('StickyTopNavbar component renders unchanged', () => {
  const tree = renderer.create(<Navigation 
    data={testData} 
    handleUpdatedCartInState={handleUpdatedCartInState} 
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
