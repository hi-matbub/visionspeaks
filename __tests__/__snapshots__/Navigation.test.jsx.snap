// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StickyTopNavbar component renders unchanged 1`] = `
<div
  className="d-flex justify-content-between w-100 fixed-top"
  style={
    Object {
      "background": "#FFFFFF !important",
    }
  }
>
  <nav
    className="justify-content-between fixed-top navbar navbar-expand-lg navbar-light fixed-top"
    style={
      Object {
        "backgroundColor": "#FFFFFF !important",
      }
    }
  >
    <img
      alt="React Bootstrap logo"
      className="d-inline-block align-top"
      height="75"
      src="/logo.svg"
    />
    <a
      className="navbar-brand"
      href="/"
    >
      Dog-Gone Goods
    </a>
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    >
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "none",
            "border": "0",
          }
        }
        type="button"
      >
        <div
          className="bg-warning scale-in-center"
          style={
            Object {
              "borderRadius": "50%",
              "height": "15px",
              "position": "absolute",
              "right": "14px",
              "top": "28px",
              "width": "15px",
            }
          }
        />
        <img
          alt="Checkout"
          src="/cart.png"
          style={
            Object {
              "cursor": "pointer",
              "height": "20px",
            }
          }
        />
      </button>
    </div>
  </nav>
</div>
`;
